pystac/item.py

```diff
@@ -384,4 +411,102 @@ def from_dict(d):
                      media_type=media_type,
                      title=title,
                      description=description,
+                     roles=roles,
                      properties=properties)
+
+
+class CommonMetadata:
```

Comment:
General comment on the implementation of CommonMetadata.

The current implementation would work well for retrieving metadata. But the API implement here for setting data could be improved. For instance I would image a user would think this would work, but it won't:

```python
item.common_metadata.license = 'something'
print(item.common_metadata.license)
> None
```

One would have to do this to property set the license:

```python
cm = item.common_metadata
cm.license = 'something'
item.set_common_metadata(cm)
print(item.common_metadata.license)
> something
```

I think there's a way to implement this property object idea while having all of the data still stored in the original item. Consider an approach where:

- Instead of the CommonMetada __init__ building up internal properties from the passed in `properties` object, just store that properties object.
- Define property getters and setters for each of the common metadata properties. The getters operate the same as the logic you use to set the properties currently in the __init__ - you return the values directly from the `properties` object. The setters then place the value directly into the properties object in the format that `to_dict` would already use. This way you set the properties directly on the property object of the item in the correct forrmat.

For example, if I replace the current implementation of CommonMetadata with this:

```python
class CommonMetadata:
    def __init__(self, properties):
        self.properties = properties

    @property
    def license(self):
        return self.properties.get('license')

    @license.setter
    def license(self, v):
        self.properties['license'] = v
```

And then run a similar set of code, I get the  expected result:

```python
item.common_metadata.license = 'something'
print(item.common_metadata.license)
> something
```

This also gets rid of the need for a `to_dict` method, as the properties are set directly into the item's properties and will be written out with the Item's `to_dict` method properly:

```python
print(item.properties)
> {'datetime': '2019-10-04T18:55:37Z', 'license': 'something'}
```

If we go this route, I would also recommend getting rid of the CommonMetadata.from_dict method - this is a sort of 'view' object that is different from other stac objects in the codebase and we wouldn't expect users to really use them directly.

Reply:
Makes a lot of sense, just took a shot at implementing this and pushed it up

Reply:
this commit https://github.com/azavea/pystac/tree/133955b146325dc4040a4aaaf7e0044549fe7ff6